[Sun Yu Ting] iP

Effixion opened the pull request.

Commit:
Add Gradle support

Effixion committed:
Added Level 1

Effixion committed:
Added TaskList and Added Level 2

Effixion committed:
Added Level 2

Effixion committed:
Added Level-3

Effixion committed:
Added Level-4

Effixion committed:
Added Automated Text UI Testing

Effixion committed:
Added exceptions and handling of exceptions

Effixion committed:
Added delete funtionality using ArrayList

Effixion committed:
Add the ability to save tasks to the hard disk. Only applicable to todo tasks for now.

Effixion committed:
Add Level-7

Effixion committed:
Add Level-8

Effixion committed:
Merge branch 'branch-Level-7'

Effixion committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/TaskList.java
Merge Level-8 branch with main branch

Effixion committed:
Fix all the merging problems

Effixion committed:
Fix some spacing issues causing the program to throw an exception

Effixion committed:
Add MoreOOP

Effixion committed:
Organise classes into different packages

Effixion committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

Effixion committed:
Add JUnit Tests for Ui and TaskList

Effixion committed:
Add JavaDocs to most classes and public methods

Effixion committed:
Add Level-9

Effixion committed:
Tweak code to comply with coding standard

Effixion committed:
Merge branch 'branch-Level-9'

Effixion committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parser/Parser.java

Effixion committed:
Update README.md

Effixion committed:
Use checkstyle to fix coding style violations

Effixion committed:
Add GUI using JavaFX and fixed checkstyle errors

Effixion committed:
Merge remote-tracking branch 'origin/master' into branch-A-CheckStyle

Effixion committed:
Merge branch 'master' into branch-A-CheckStyle

Effixion committed:
Update README

Effixion committed:
Merge branch 'branch-A-CheckStyle'

lilythchu commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,74 @@
+public class Storage {
+
+    private static String filePath;
+    private File f;
```
Maybe you could name this variable more specifically. As said in the textbook, a name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail.

sunyuting2000 replied:
Okay thanks for your recommendation!

lilythchu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,167 @@
+        String[] splitCommand = s.split(" ", 2);
+        String command = splitCommand[0];
+        switch (command) {
+        case "bye":
```
This method is quite long. Maybe you could split it into some small methods for each case.

sunyuting2000 replied:
I agree with your criticism here. I will try to reduce the method length when I do have the time to look at this again.

lilythchu reviewed (commented):
Your code looks good! I feel that your code was easy to read as it was very organized.

yeoyujie commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,167 @@
+ */
+public class Parser {
+
+    private static TaskList taskList;
```
Should we extract out tasklist storage and UI? Could consider using Duke.java to control the logic flow instead of exposing storage, UI and tasklist all to parser?

sunyuting2000 replied:
This is true since I am indeed exposing these classes to parser. I will try to address this issue when I have the time.

yeoyujie commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,74 @@
+     * Reads the tasks from the file and loads them into an <code>ArrayList</code>
+     * @return An <code>ArrayList</code> containing the tasks from the file.
+     */
+    public ArrayList<String> load() {
```
Could consider renaming this loadFromFile to standardise with appendToFile

sunyuting2000 replied:
Thank you for the suggestion. I will change it soon.

yeoyujie reviewed (commented):
LGTM, overall this seem v solid.

Effixion committed:
Update build.gradle to generate JAR files

Effixion committed:
Add a goodbye message when the user enters bye.

Effixion committed:
Add assertions. Update the names in Storage
to fit naming conventions.

Assertions were placed in the readInputString method in Parser
so that the first element in splitCommand will never be null. This
is important to check since the rest of the code depends on
what is contained in the first element of splitCommand.

Also, comments made by my peers about the naming of certain
variables and methods were heeded. I changed the name of load()
to loadFromFile() and I changed the name of File f to File taskFile.

Effixion committed:
Add assertions. Update the names in Storage to fit naming conventions.

Effixion committed:
Improve code quality through refactoring.

Effixion committed:
Improve code quality through refactoring.

The "bye" case in readInputString method was not adhering to
SLAP.

Thus, I added in two methods into taskList so that I could abstract
out what I was doing before so as to SLAP hard. This would make
it easier to understand the code.

Also, since I had not fully implemented changing of the date format,
I practiced KISSing. I made the code simple to change the date
format. I also added in new methods in the UI to show related
messages when the user want to change the date format.

Effixion committed:
Revert "Improve code quality through refactoring."

This reverts commit 8f15150142b62c1c91d9f1bf824d99d5632294a0.

Effixion committed:
Revert "Add assertions. Update the names in Storage to fit naming conventions."

This reverts commit aeb09295297cea424a407c036f869919fd05f18f.

Effixion committed:
Merge branch 'branch-A-Assertions' of https://github.com/Effixion/ip into branch-A-Assertions

Effixion committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/Effixion/ip into branch-A-CodeQuality

Effixion committed:
Merge pull request #2 from Effixion/branch-A-Assertions

Add Assertions

Effixion committed:
Merge pull request #3 from Effixion/branch-A-CodeQuality

Improve Code Quality

Effixion committed:
Add C-Tagging.

Users now have the ability to tag the tasks that they add.
They can also choose not to tag the task as well.
All tags will be prefixed by a hash symbol, i.e. #.
This will allow users to customize their tasks to a greater
extent.

Effixion committed:
Merge branch 'branch-C-Tagging'

Effixion committed:
Update README.md

Effixion committed:
Add abstract class Command. Fix some formatting issues when
reading and writing to the external file. Add a findtag command for
the user to find a tag in their task list.

Effixion committed:
Merge branch 'master' of https://github.com/Effixion/ip

Effixion committed:
Add A-Personality

Effixion committed:
Add representative screenshot

Effixion committed:
Update README.md

Effixion committed:
Add more exception handling

Effixion committed:
Change build.gradle for .jar

Effixion committed:
Fixed main class name in build.gradle
